File: src/Domain/Entities/ValorEsperado.cs
using System;
using System.Collections.Generic;

namespace Domain.Entities
{
    public class ValorEsperado
    {
        public int IdValorEsperado { get; private set; }
        public string Descricao { get; private set; }
        public string Valor { get; private set; }
        public int BancoId { get; private set; }
        public int TipoCNABId { get; private set; }
        public int TipoBoletoId { get; private set; }
        public DateTime DataCadastro { get; private set; }
        public DateTime? DataAlteracao { get; private set; }
        public bool Ativo { get; private set; }

        public Banco Banco { get; private set; }
        public List<LayoutValorEsperado> LayoutValoresEsperados { get; private set; }

        internal void SetarDataAlteracao()
        {
            DataAlteracao = DateTime.Now;
        }

        internal void AlterarDados(string descricao, string valor, int bancoId, int tipoCNABId, int tipoBoletoId, bool ativo)
        {
            Descricao = descricao;
            Valor = valor;
            BancoId = bancoId;
            TipoCNABId = tipoCNABId;
            TipoBoletoId = tipoBoletoId;
            Ativo = ativo;
        }

        internal void SetarDataCadastro(DateTime now)
        {
            DataCadastro = now;
        }

        internal void LimparEntidades()
        {
            Banco = null;
        }
    }
}

File: src/APP/src/app/telas-adm/valor-esperado-banco/model/valor-esperado.ts
import { LayoutValorEsperado } from "./layout-valor-esperado";
import { Banco } from "../../banco/models/banco";

export class ValorEsperado{
    
    constructor() {
        
        this.banco = new Banco();
    }

    idValorEsperado: number;
    descricao: string;
    valor: string;
    bancoId: number;
    tipoCNABId: number;
    tipoBoletoId: number;
    dataCadastro: Date;
    ativo: boolean;
    banco?: Banco;
    cnabDescricao: string;
    tipoBoletoDescricao: string;
    layoutValoresEsperados?: Array<LayoutValorEsperado>;
}